generator/types.go

```diff
@@ -637,6 +651,7 @@ func (t *typeResolver) ResolveSchema(schema *spec.Schema, isAnonymous, isRequire
 
 	tpe := t.firstType(schema)
 	defer setIsEmptyOmitted(&result, schema, tpe)
+	defer setIsJsonString(&result, schema, tpe)
```

Comment:
There is a known bug with extensions resolve in the case of $ref. See #2148 to fix the defer statement for this case.